web-portal/store/venues.js

```diff
@@ -0,0 +1,143 @@
+import { ApiService } from '../services/ApiService'
+import { initialQueryState, setQueryFetching, setQueryStateFail, setQueryStateSuccess } from './utils/queryState'
+
+export const state = () => {
+  return {
+    getActiveVenuesQuery: initialQueryState(),
+    getDeletedVenuesQuery: initialQueryState(),
+    deleteVenues: initialQueryState(),
+    activateVenueQuery: initialQueryState()
+  }
+}
+
+export const getters = {
+  GetAllActiveVenues: (state) => {
+    return state.getActiveVenuesQuery
+  },
+  GetDeletedVenues: (state) => {
+    return state.getDeletedVenuesQuery
+  }
+}
+
+export const mutations = {
+  ACTIVE_VENUES_FETCH_START: (state) => {
+    setQueryFetching(state.getActiveVenuesQuery)
+  },
+  ACTIVE_VENUES_FETCH_SUCCESS: (state, payload) => {
+    setQueryStateSuccess(state.getActiveVenuesQuery, payload)
+  },
+  ACTIVE_VENUES_FETCH_FAIL: (state, payload) => {
+    setQueryStateFail(state.getActiveVenuesQuery, payload)
+  },
+
+  DELETED_VENUES_FETCH_START: (state) => {
+    setQueryFetching(state.getDeletedVenuesQuery)
+  },
+  DELETED_VENUES_FETCH_SUCCESS: (state, payload) => {
+    setQueryStateSuccess(state.getDeletedVenuesQuery, payload)
+  },
+  DELETED_VENUES_FETCH_FAIL: (state, payload) => {
+    setQueryStateFail(state.getDeletedVenuesQuery, payload)
+  },
+
+  VENUE_DELETE_START: (state) => {
+    setQueryFetching(state.deleteVenues)
+  },
+  VENUE_DELETE_SUCCESS: (state, { id, venue }) => {
+    setQueryStateSuccess(state.deleteVenues, venue)
+
+    if (state.getActiveVenuesQuery.isSuccess && !state.getActiveVenuesQuery.isFetching) {
+      state.getActiveVenuesQuery.data = state.getActiveVenuesQuery.data.filter(venue => venue.id !== id)
+    }
+
+    if (state.getDeletedVenuesQuery.isSuccess && !state.getDeletedVenuesQuery.isFetching) {
+      state.getDeletedVenuesQuery.data.push(venue)
+    }
+  },
+  VENUE_DELETE_FAIL: (state, payload) => {
+    setQueryStateFail(state.deleteVenues, payload)
+  },
+
+  VENUE_ACTIVATE_START: (state) => {
+    setQueryFetching(state.activateVenueQuery)
+  },
+  VENUE_ACTIVATE_SUCCESS: (state, { id, venue }) => {
+    setQueryStateSuccess(state.activateVenueQuery, venue)
+    if (state.getActiveVenuesQuery.isSuccess && !state.getActiveVenuesQuery.isFetching) {
+      state.getActiveVenuesQuery.data.push(venue)
+    }
+
+    if (state.getDeletedVenuesQuery.isSuccess && !state.getDeletedVenuesQuery.isFetching) {
+      state.getDeletedVenuesQuery.data = state.getDeletedVenuesQuery.data.filter(venue => venue.id !== id)
+    }
+  },
+  VENUE_ACTIVATE_FAIL: (state, payload) => {
+    setQueryStateFail(state.activateVenueQuery, payload)
+  }
+}
+
+export const actions = {
+  FetchActiveVenues: (context) => {
+    context.commit('ACTIVE_VENUES_FETCH_START')
+
+    return ApiService.get('/venues?includeDeleted=no')
```

Comment:
I may be misunderstanding this but my expectation would be that plain `/venues` would only return non-soft deleted venues by default, and require a flag to do anything else.

Reply:
yeah just being explicit, but this is not needed, the server default to no